validator/valnode/redis/consumer.go

```diff
@@ -42,12 +45,56 @@ func NewValidationServer(cfg *ValidationServerConfig, spawner validator.Validati
 		}
 		consumers[mr] = c
 	}
+	var (
+		wg          sync.WaitGroup
+		initialized atomic.Bool
+	)
+	initialized.Store(true)
+	for i := 0; i < len(cfg.ModuleRoots); i++ {
+		mr := cfg.ModuleRoots[i]
+		wg.Add(1)
+		go func() {
+			defer wg.Done()
+			done := waitForStream(redisClient, mr)
+			select {
+			case <-time.After(cfg.StreamTimeout):
+				initialized.Store(false)
```

Comment:
this is the other way around.
We should only post an error if no wasm-module-root connected. If only one connected that's enough.